how-to-delete-browser-cookies/script.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * The list of cookies to keep
+ */
+(function () {
+  var itemsToKeep = ["euconsent-v2", "didomi_token"];
+  var itemsToKeepRegex = /some_regex_[a-z0-9]*/;
+
+  /**
+   * Returns cookie value
+   */
+  var getCookieValue = function (cookieName) {
+    var cookie = document.cookie.split(";").filter(function (cookieValue) {
+      return cookieValue.indexOf(cookieName) !== -1;
+    })[0];
+    if (cookie) {
+      return cookie.split("=")[1];
+    }
+  };
+
+  var deleteCookie = function (name, domain, path) {
+    path = path || "/";
+    var cookie = [
+      name + "=",
+      "expires=Thu, 01 Jan 1970 00:00:01 GMT",
+      "path=" + path,
+    ];
+    if (domain) {
+      cookie.push("domain=" + domain);
+    }
+    document.cookie = cookie.join(";");
+  };
+
+  /**
+   * Check if all vendor and purposes are disabled
+   */
+  var areAllVendorsAndPurposesDisabled = function () {
+    var enabledEntities = [];
+    var disabledEntities = [];
+    var data = window.Didomi.getUserStatus();
+
+    data.vendors.consent.enabled.forEach(function (entity) {
+      enabledEntities.push(entity);
+    });
+
+    data.purposes.consent.enabled.forEach(function (entity) {
+      enabledEntities.push(entity);
+    });
+
+    data.vendors.consent.disabled.forEach(function (entity) {
+      disabledEntities.push(entity);
+    });
+
+    data.purposes.consent.disabled.forEach(function (entity) {
+      disabledEntities.push(entity);
+    });
+
+    /**
+     * We check that we don't have any enabled entities
+     * and that disabled entities are present
+     */
+    return enabledEntities.length === 0 && disabledEntities.length > 0;
```

Comment:
minor performance suggestion here, but couldn't you just do 4 length checks and skip the 4 loops (e.g., `return (data.vendors.consent.enabled.length + data.purposes.consent.enabled.length) === 0 && (data.vendors.consent.disabled.length + data.purposes.consent.disabled.length) > 0`) or store those sums in vars for readability?